tests/toolkit/matl_buy_policy_tests.cc

```diff
@@ -113,17 +135,75 @@ TEST_F(MatlBuyPolicyTests, Reqs) {
   }
   ASSERT_FALSE(req->exclusive());
   ASSERT_FLOAT_EQ(req->target()->quantity(), cap);
+}
+
+// Test single quantized request
+TEST_F(MatlBuyPolicyTests, Quantize) {
+  double cap = 5;
+  ResBuf<Material> buff;
+  buff.capacity(cap);
+  double p2 = 2.5;
+  cyclus::Composition::Ptr c1 = cyclus::Composition::Ptr(new TestComp()); 
+  MatlBuyPolicy p;
+
+  double quantize = 2.5;
+  p.Init(fac1, &buff, "", std::numeric_limits<double>::max(), 1, 1, quantize).Set("commod1", c1);
+  std::set<RequestPortfolio<Material>::Ptr> obs = p.GetMatlRequests();
+  ASSERT_EQ(obs.size(), 2);
+  ASSERT_EQ((*obs.begin())->requests().size(), 1);
+  Request<Material>* req = (*obs.begin())->requests().at(0);
+  ASSERT_TRUE(req->exclusive());
+  ASSERT_FLOAT_EQ(req->target()->quantity(), quantize);
+}
+
+// Test two quantized requests
+TEST_F(MatlBuyPolicyTests, MultiReqQuantize) {
+    double cap = 5;
+  ResBuf<Material> buff;
+  buff.capacity(cap);
+  cyclus::Composition::Ptr c1 = cyclus::Composition::Ptr(new TestComp()); 
+  cyclus::Composition::Ptr c2 = cyclus::Composition::Ptr(new TestComp()); 
+  MatlBuyPolicy p;
   
   // two portfolios with quantize
   double quantize = 2.5;
-  p.Init(fac1, &buff, "", std::numeric_limits<double>::max(), 1, 1, quantize);
-  obs = p.GetMatlRequests();
+  p.Init(fac1, &buff, "", std::numeric_limits<double>::max(), 1, 1, quantize).Set("commod1", c1).Set("commod2", c2);
+  std::set<RequestPortfolio<Material>::Ptr> obs = p.GetMatlRequests();
   ASSERT_EQ(obs.size(), 2);
   ASSERT_EQ((*obs.begin())->requests().size(), 2);
   ASSERT_EQ((*(obs.begin()++))->requests().size(), 2);
-  req = (*obs.begin())->requests().at(0);
+  Request<Material>* req = (*obs.begin())->requests().at(0);
   ASSERT_TRUE(req->exclusive());
   ASSERT_FLOAT_EQ(req->target()->quantity(), quantize);
+
+  req = (*(obs.begin()++))->requests().at(0);
+  ASSERT_TRUE(req->exclusive());
+  ASSERT_FLOAT_EQ(req->target()->quantity(), quantize);
+}
+
+// Tests that agent cycles between active and dormant cycles during a mock
+// sim, only buying when active.
+TEST_F(MatlBuyPolicyTests, ActiveDormant) {
+  int active = 2;
+  int dormant = 3;
+  int dur = 4;
+  double throughput = 1;
+
+  cyclus::MockSim sim(dur);
+  cyclus::Agent* a = new TestFacility(sim.context());
```

Comment:
For cleanliness, you should `delete a` at the end of this test.  (it doesn't really matter because it's a test, but best not to be in the practice of leaving memory leaks around)

Reply:
added